packages/repository/src/decorators/repository.ts

```
@@ -104,11 +105,27 @@ export function repository<T extends Model>(
       } else {
         // Use repository-factory to create a repository from model + dataSource
         // inject('repository-factory', meta)(target, key!, descriptor);
-        throw new Error('@repository(model, dataSource) is not implemented');
+        inject('', meta, resolve)(target, key!, descriptor);
+        // throw new Error('@repository(model, dataSource) is not implemented');
       }
       return;
     }
     // Mixin repostory into the class
     throw new Error('Class level @repository is not implemented');
   };
 }
+
+async function resolve(ctx: Context, injection: Injection) {
+  const meta = injection.metadata as RepositoryMetadata;
+  let modelClass = meta.modelClass;
+  if (meta.modelName) {
+    modelClass = await ctx.get('models.' + meta.modelName);
+  }
+  let dataSource = meta.dataSource;
+  if (meta.dataSourceName) {
+    dataSource = await ctx.get('datasources.' + meta.dataSourceName);
+  }
+  return new DefaultCrudRepository(
+    modelClass as typeof juggler.PersistedModel,
+    dataSource! as juggler.DataSource);
```

What if `dataSource` is null? Shouldn't we report a helpful error in such case?